src/endpoints/auth/auth.controller.ts

```diff
@@ -0,0 +1,128 @@
+
+const ajv = new Ajv();
+
+const { secret, accessDeathDate, refreshDeathDate } = jwtConfig;
+const { getUser } = LocalStorage;
+
+export const signUpRouter: FastifyPluginAsync<FastifyPluginOptions> = async (server, options) => {
+  server.post<{ Body: IBodySignUp }>(
+    '/sign-up',
+    {
+      schema: {
+        tags: ['auth'],
+        summary: 'Sign up',
+        body: {
+          type: 'object',
+          properties: {
+            email: { type: 'string', minLength: 6, maxLength: 256, example: 'only@test.com' },
+            password: {
+              type: 'string',
+              minLength: 8,
+              maxLength: 256,
+              example: 'passwordTest',
+            },
+          },
+          required: ['email', 'password'],
+        },
+        response: {
+          200: signUpOutputSchema,
+        },
+      },
+    },
+    async (req, reply) => {
+      const { email, password } = req.body;
+
+      // todo 28.11.2021: remove later - only here for testing
+      await UserEntity.delete({ email });
+
+      const user = await UserEntity.findOne({ email });
+
+      if (user) {
+        throw createError(400, 'User exists.');
+      }
+
+      const sessionKey = randomUUID();
+      const salt = randomUUID();
+      const passwordHash = HashingService.createHash(password, sessionKey);
+      const dataUser = UserEntity.create({ email, passwordHash });
+
+      const { accessToken, refreshToken, accessTokenExpiresIn, refreshTokenExpiresIn } = await JWTService.generateTokens({
+        sessionKey,
+        jwtSecret: secret,
+        accessDeathDate,
+        refreshDeathDate,
+      });
+
+      dataUser.sessionKey = sessionKey;
+      dataUser.salt = salt;
+      await dataUser.save();
+
+      const { code, expiresAt } = VerificationCodeService.createCode();
+
+      const dataCodes = VerificationCodesEntity.create({ userId: dataUser._id, code, expiresAt });
+      await dataCodes.save();
+
+      await EmailService.sendMessageToEmail(email, code);
+
+      return { accessToken, refreshToken, accessTokenExpiresIn, refreshTokenExpiresIn };
+    }
+  );
+};
+
+export const validateEmailRouter: FastifyPluginAsync<FastifyPluginOptions> = async (server, options) => {
+  server.post<{ Body: IBodyValidateEmail; Headers: IHeadersValidateEmail }>(
+    '/email/validate',
+    {
+      schema: {
+        tags: ['auth'],
+        summary: 'Verify email here',
+        body: {
+          type: 'object',
+          properties: {
+            code: { type: 'string', minLength: 6, maxLength: 6 },
+          },
+          required: ['code'],
+        },
+        headers: {
+          type: 'object',
+          properties: {
+            authorization: { type: 'string' },
+          },
+          required: ['authorization'],
+        },
+        response: {
+          200: valitdateEmailOutputSchema,
+        },
+      },
+    },
+    async (req, reply) => {
+      const { code } = req.body;
+      const user = getUser();
```

Comment:
Don't do this. It's unclear where did this method come from, it looks like a native method for this module but it isn't. Do `LocalStorage.getUser()`